### PR TITLE
Chat only shows groups joined by user

### PR DESCRIPTION
**Pull request by vlop11**
Small fix that changed the groups query in ChatServlet to only show groups joined by the user

**I'm closing this PR and instead merging it with https://github.com/googleinterns/step123-2020/pull/25**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/GroupsServlet.java [hunk 1]
```diff
@@ -57,18 +58,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
         ImmutableList<ImmutableMap<String, String>> groupsList = groupsListBuilder.build();
+        ImmutableList<Long> userGroups = ImmutableList.copyOf(
+            ServletUtils.getGroupIdList(request.getUserPrincipal().getName()));
         
         // Each group has its own map which points to its info and all maps are passed into the template as a list
         // This will make it easier when groups are queried from Datastore
-        ImmutableMap<String, ImmutableList<ImmutableMap<String, String>>> groupsData = 
-            ImmutableMap.of(GROUPS_KEY, groupsList);
+        ImmutableMap<String, ImmutableList> groupsData = 
+            ImmutableMap.of(GROUPS_KEY, groupsList, USER_GROUPS, userGroups);
```
Comment (ahlschlager) [inline src/main/java/com/google/sps/servlets/GroupsServlet.java:76]: Is this the map of the groups from getGroupsList(userEmail) or is this the new function you created?

File: src/main/java/com/google/sps/servlets/GroupsServlet.java [hunk 2]
```diff
@@ -57,18 +58,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         ImmutableList.Builder<ImmutableMap<String, String>> groupsListBuilder = new ImmutableList.Builder<>();
```
Comment (ahlschlager) [inline src/main/java/com/google/sps/servlets/GroupsServlet.java:58]: I can't leave at comment at unchanged code, but at the top it says TODO: Unhardcode the groups, do you want to leave that hardcoded group as the default group that everyone is automatically join to?

File: src/test/java/com/google/sps/ChatServletTest.java [hunk 3]
```diff
@@ -115,161 +121,161 @@ public void testsCommentScore() throws IOException {
         Assert.assertTrue(actualScore <= max);
     }
 
-    @Test
-    public void servletGetNoComments() throws IOException {
-        // GET method is called and should respond with the HTML
-        // string of the chat template with no comments
+    // @Test
```
Comment (ahlschlager) [inline src/test/java/com/google/sps/ChatServletTest.java]: Did you comment out all the tests?

File: src/main/java/com/google/sps/servlets/GroupsServlet.java (hunk 2 above)
Comment (dylancastillo) [reply src/main/java/com/google/sps/servlets/GroupsServlet.java:58]: i removed it in my PR so it should be dealt with once it's merging!

File: src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 4]
```diff
@@ -47,33 +48,20 @@
...
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-
-        Query groupQuery = new Query(GROUP_KIND);
-        PreparedQuery preparedGroupQuery = datastore.prepare(groupQuery);
-
-        ImmutableList.Builder<ImmutableMap<String, String>> groupsListBuilder = new ImmutableList.Builder<>();
-        for (Entity group : preparedGroupQuery.asIterable()) {
-            if (defaultGroup == null) {
```
Comment (dylancastillo) [inline src/main/java/com/google/sps/servlets/ChatServlet.java:57]: you're not changing defaultGroup anymore so it'll always be null, and doPost seems to use it. not sure if that would be an issue

File: src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 5]
```diff
@@ -47,33 +48,20 @@
...
-        }
-        ImmutableList<ImmutableMap<String, String>> groupsList = groupsListBuilder.build();
+        ImmutableList<ImmutableMap<String, String>> groupsList = 
+            ServletUtils.getGroupsList(request.getUserPrincipal().getName());
 
         String groupId = (String) request.getParameter(GROUP_ID_PROPERTY);
         if (groupsList.isEmpty()) {
             // If there are no groups, then the current group is just an empty ID
             groupId = "";
         } else if (groupId == null || groupId.isEmpty()) {
```
Comment (dylancastillo) [inline src/main/java/com/google/sps/servlets/ChatServlet.java]: do Strings.isNullOrEmpty(groupId)

File: src/main/java/com/google/sps/servlets/GroupsServlet.java (hunk 1 above)
Comment (vlop11) [reply src/main/java/com/google/sps/servlets/GroupsServlet.java:76]: It's from the new function I made. basically it's just a list of all the group Ids the user has joined

File: src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 6]
```diff
@@ -47,33 +49,24 @@
...
-            }
-
-            ImmutableMap<String, String> groupMap = ImmutableMap.of(
-                GROUP_NAME_PROPERTY, (String) group.getProperty(GROUP_NAME_PROPERTY),
-                GROUP_ID_PROPERTY, group.getKey().getName());
-            groupsListBuilder.add(groupMap);
-        }
-        ImmutableList<ImmutableMap<String, String>> groupsList = groupsListBuilder.build();
+        // ImmutableList<ImmutableMap<String, String>> groupsList = 
+        //     ServletUtils.getGroupsList(request.getUserPrincipal().getName());
```
Comment (ahlschlager) [inline src/main/java/com/google/sps/servlets/ChatServlet.java:53]: Remove these if your not going to use them. Unless your using the sample in the next lines

File: src/main/java/com/google/sps/utils/ServletUtils.java [hunk 7]
```diff
@@ -63,4 +66,57 @@ public static void enforceUserLogin(HttpServletRequest request, HttpServletRespo
...
+      if (datastoreList != null) {
+        // If the user has joined groups, get those instead
+        userGroups = ImmutableList.copyOf(datastoreList);
+      }
+    } else {
+      // if the user does not exist, create one and add to datastore
+      user = new Entity(USER_KIND, userEmail);
+
+      user.setProperty(USER_EMAIL_PROPERTY, userEmail);
+      user.setProperty(GROUPS_KEY, userGroups);
```
Comment (ahlschlager) [inline src/main/java/com/google/sps/utils/ServletUtils.java:91]: Were you going to make a method that checks if the user exists?

**Comment by vlop11** [comment]
I'm closing this PR and instead merging it with [https://github.com/googleinterns/step123-2020/pull/25](url)

**Pull request closed without merging**